### PR TITLE
Fix smartStep in ignored sourcemaps, add test

### LINKED ISSUE #109: smartStep incorrectly steps through "ignored" files
- Set a bp in a compiled file
- The sourcemap is temporarily ignored when hitting the BP
- But smartStep steps through this file

### PR DESCRIPTION
**Pull request by roblourens**
Fixes #109

### CODE REVIEW

File: src/adapter/sources.ts [hunk 1]
```diff
@@ -355,29 +369,29 @@ export class SourceContainer {
...
+    if (!isUiLocation(sourceMapUiLocation))
       return [];
 
     const r = this.getSourceMapUiLocations(sourceMapUiLocation);
     r.push(sourceMapUiLocation);
     return r;
   }
 
-  _sourceMappedUiLocation(uiLocation: UiLocation, map: SourceMap): UiLocation | undefined {
+  _sourceMappedUiLocation(uiLocation: UiLocation, map: SourceMap): UiLocation | UnmappedReason {
```
Comment (roblourens) [inline src/adapter/sources.ts]: This is awkward, let me know if you have a better idea

File: src/adapter/sources.ts [hunk 2]
```diff
@@ -23,7 +23,13 @@ export interface UiLocation {
   lineNumber: number; // 1-based
   columnNumber: number;  // 1-based
   source: Source;
-};
+}
+
+function isUiLocation(loc: any): loc is UiLocation {
```
Comment (connor4312) [inline src/adapter/sources.ts]: `unknown` would be the more correct type to use here, TS will force you to narrow it safely :)

File: src/adapter/sources.ts (hunk 1 above)
Comment (connor4312) [reply src/adapter/sources.ts]: I think this is fine. Throwing some typed error would be traditional OO thing to do but I don't mind this; it forces consumers to handle failures explicitly.

File: src/adapter/sources.ts [hunk 3]
```diff
@@ -233,8 +239,14 @@ export class Source {
...
 
-export interface PreferredUILocation extends UiLocation {
+export interface PreferredUiLocation extends UiLocation {
   isMapped: boolean;
+  unmappedReason?: UnmappedReason;
+}
+
+export enum UnmappedReason {
+  MapDisabled,
+  CannotMap
```
Comment (connor4312) [inline src/adapter/sources.ts]: Docblocks may be useful here for future readers

File: src/adapter/sources.ts (hunk 1 above)
Comment (roblourens) [reply src/adapter/sources.ts]: Yeah and since I need to keep the failure type around later on PreferredUiLocation, I wanted to just introduce it here.

**Pull request merged**